src/powersystem.jl

```diff
@@ -60,15 +62,194 @@ struct PowerNetwork
+    - VM for buses of type SLACK (type 3)
+These variables are controlled by the grid operator.
+
+Vector p is the parameter vector, consisting on:
+    - VA for buses of type SLACK (type 3)
+    - P, Q for buses of type PQ (type 1)
+These parameters are fixed through the problem.
+
+In addition, variables:
+    - Q for buses of type PV (type 2)
+    - P, Q for buses of type SLACK (type 3)
+are explicitly obtained from the previous variables, and hence
+not needed in the computation.
+"""
+function assemble_vecs(pf::PowerNetwork)
+ 
+    nref = length(pf.ref)
+    npv = length(pf.pv)
+    npq = length(pf.pq)
+    println(nref, npv, npq)
+
+
+    # build vector x
+    dimension = 2*npq + npv
+    x = zeros(dimension, 1)
+    k = 1
+
+    for bus=1:pf.nbus
+        if pf.bustype[bus] == 1 #PQ
+            x[k] = abs(pf.V[bus])
+            x[k + 1] = angle(pf.V[bus])
+            k = k + 2
+        elseif pf.bustype[bus] == 2 #PV
+            x[k] = angle(pf.V[bus])
+            k = k + 1
+        end
+    end
+
+    # build vector u
+    dimension = 2*npv + nref
+    u = zeros(dimension, 1)
+    k = 1
+
+    for bus=1:pf.nbus
+        if pf.bustype[bus] == 2 #PV
+            u[k] = abs(pf.V[bus])
+            u[k + 1] = real(pf.Sbus[bus])
+            k = k + 2
+        elseif pf.bustype[bus] == 3 # slack
+            u[k] = abs(pf.V[bus])
+            k = k + 1
+        end
+    end
+
+    # build vector p
+    dimension = nref + 2*npq
+    p = zeros(dimension, 1)
+    k = 1
+    
+    for bus=1:pf.nbus
+        if pf.bustype[bus] == 3 #slack
+            p[k] = abs(pf.V[bus])
+            k = k + 1
+        elseif pf.bustype[bus] == 1 # PQ
+            p[k] = real(pf.Sbus[bus])
+            p[k + 1] = imag(pf.Sbus[bus])
+            k = k + 2
+        end
+    end
+
+    return x, u, p
+
+end
+
+
+"""
+    xup_to_vs(PowerNetwork, x, u, p)
+
+Converts x, u, p vectors to vmag, vang, pinj and qinj.
+"""
+function xup_to_vs(pf::PowerNetwork, x::Array{Float64}, u::Array{Float64},
+                   p::Array{Float64})
+
+    nbus = pf.nbus
+    nref = length(pf.ref)
+    npv = length(pf.pv)
+    npq = length(pf.pq)
+
+    vmag = zeros(nbus)
+    vang = zeros(nbus)
+    pinj = zeros(nbus)
+    qinj = zeros(nbus)
+
+    x_idx = 1
+    u_idx = 1
+    p_idx = 1
+
+    for bus=1:nbus
+        if pf.bustype[bus] == 1 #PQ
+            vmag[bus] = x[x_idx]
+            vang[bus] = x[x_idx + 1]
+            pinj[bus] = p[p_idx]
+            qinj[bus] = p[p_idx + 1]
+            x_idx += 2
+            p_idx += 2
+        elseif pf.bustype[bus] == 2 #PV
+            vmag[bus] = u[u_idx]
+            pinj[bus] = u[u_idx + 1]
+            vang[bus] = x[x_idx]
+            u_idx += 2
+            x_idx += 1
+        elseif pf.bustype[bus] == 3 # ref
+            vmag[bus] = u[u_idx]
+            vang[bus] = p[p_idx]
+            u_idx += 1
+            p_idx += 1
+        end
+
+    end
+
+
+    return vmag, vang, pinj, qinj
+
+end
+
+
```

Comment:
I would be in favor of having only one blank line between two functions

Reply:
@frapac sorry Ill pay more attention to the style guide in the future.